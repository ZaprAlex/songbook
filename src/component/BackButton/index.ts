export { default } from './BackButton';
export * from './types';
