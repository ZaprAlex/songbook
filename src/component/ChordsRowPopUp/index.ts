import ChordsRowPopUp from './ChordsRowPopUp';

export default ChordsRowPopUp;
