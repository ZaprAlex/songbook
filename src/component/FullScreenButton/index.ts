export { default } from './FullScreenButton';
export * from './types';
