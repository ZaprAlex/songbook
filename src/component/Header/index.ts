export { default } from './Header';
export * from './types';
