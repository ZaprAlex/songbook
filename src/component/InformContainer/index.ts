export { default } from './InformContainer';
export * from './types';
