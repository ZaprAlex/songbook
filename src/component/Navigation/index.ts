import NavigationProvider from './NavigationProvider';
import useAppNavigation, { AppNavigationCtx } from './NavigationContext';

export { NavigationProvider, AppNavigationCtx, useAppNavigation };
