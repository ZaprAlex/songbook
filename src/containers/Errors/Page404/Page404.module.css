.page {
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    display: flex;
    justify-content: center;
    align-items: center;
}

.content {
    display: flex;
    flex-flow: column nowrap;
    align-items: center;
    margin-top: -100px;
}

.button {
    margin-top: 24px;
}
