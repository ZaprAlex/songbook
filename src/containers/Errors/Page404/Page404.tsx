import React from 'react';

import { IButton } from '../../../component/Button';
import { useAppNavigation } from '../../../component/Navigation';
import InformContainer from '../../../component/InformContainer';
import icon from '../../../static/images/empty.png';

import styles from '../Errors.module.css';

const Page404: React.FC = () => {
    const { goToSongs } = useAppNavigation();

    const buttonProps: IButton = { size: 'x-large' };

    return (
        <div className={styles.page}>
            <InformContainer
                icon={icon}
                header="Страница не найдена"
                message="Попробуйте перейти на главную"
                buttonLabel="На главную"
                onClick={() => goToSongs()}
                contentClassName={styles.content}
                buttonClassName={styles.button}
                buttonProps={buttonProps}
            />
        </div>
    );
};

export default Page404;
