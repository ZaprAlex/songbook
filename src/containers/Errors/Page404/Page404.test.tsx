import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';

import { useAppNavigation } from '../../../component/Navigation';

import Page404 from './Page404';

jest.mock('../../../component/Navigation');

describe('Page404', () => {
    beforeEach(() => {
        jest.clearAllMocks();
        jest.restoreAllMocks();
    });

    const goToSongs = jest.fn();

    (useAppNavigation as jest.Mock).mockReturnValue({ goToSongs });

    test('should render', async () => {
        const { container } = render(<Page404 />);

        expect(container.firstChild).toMatchSnapshot();

        fireEvent.click(screen.getByRole('button'));

        expect(goToSongs).toHaveBeenCalled();
    });
});
