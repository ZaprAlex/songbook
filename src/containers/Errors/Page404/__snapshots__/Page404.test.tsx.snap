// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page404 should render 1`] = `
<div
  class="page"
>
  <div
    class="content content"
  >
    <div
      class="iconWrapper"
    >
      <img
        alt=""
        class="icon"
        data-test-id="inform-container-icon"
        src="cat-empty.png"
      />
    </div>
    <div
      class="header"
    >
      Страница не найдена
    </div>
    <div
      class="message"
    >
      Попробуйте перейти на главную
    </div>
    <div
      class="buttonBlock"
    >
      <button
        class="btn btn-green btn-contained btn-x-large btnSkeuomorph button"
        data-test-id="button-submit"
        type="button"
      >
        <span
          class="buttonLabel"
        >
          На главную
        </span>
      </button>
    </div>
  </div>
</div>
`;
